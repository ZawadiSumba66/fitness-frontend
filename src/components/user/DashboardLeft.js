import { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect, useDispatch } from 'react-redux';
import SideBar from './SideBar';
import { fetchUser } from '../../actions/user_action';

const DashboardLeft = ({ user }) => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchUser());
  }, []);

  if (!user) {
    return <h1>Loading...</h1>;
  }
  return (
    <div>
      <div className="sidebar-wrapper">
        <SideBar
          pageWrapId="page-wrap"
          outerContainerId="outer-container"
          username={user.username}
          image={user.image}
        />
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  user: state.userReducer.user.user,
});

DashboardLeft.propTypes = {
  user: PropTypes.instanceOf(Array).isRequired,
};

export default connect(mapStateToProps, null)(DashboardLeft);
