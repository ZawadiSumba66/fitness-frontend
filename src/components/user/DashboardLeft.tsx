import React, { useEffect } from 'react';
import { Spinner } from 'react-bootstrap';
import { connect, useDispatch } from 'react-redux';
import SideBar from './SideBar';
import { fetchUser, UserSignup } from '../../actions/user_action';

type UserDashboardLeft = {
  avatar: any
  user: {
    username: string,
  }
};

const DashboardLeft: React.FunctionComponent<any> = ({ user, avatar }: UserDashboardLeft) => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchUser());
  }, []);
  if (!user) {
    return (
      <div className="d-flex justify-content-center align-items-center spinner__wrapper">
        <Spinner animation="border" variant="primary" role="status">
          <span className="sr-only">Loading...</span>
        </Spinner>
      </div>
    );
  }
  return (
    <div>
      <div className="sidebar-wrapper">
        <SideBar
          pageWrapId="page-wrap"
          outerContainerId="outer-container"
          username={user.username}
          image={avatar}
        />
      </div>
    </div>
  );
};

type UserState = {
  userReducer: {
    user: {
      user: UserSignup,
      avatar_url: any,
    }
  }
};

const mapStateToProps = (state: UserState) => ({
  user: state.userReducer.user.user,
  avatar: state.userReducer.user.avatar_url,
});

export default connect(mapStateToProps, null)(DashboardLeft);
