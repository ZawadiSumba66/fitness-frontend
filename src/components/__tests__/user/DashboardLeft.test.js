import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../../store';
import DashboardLeft from '../../user/DashboardLeft';

it('matches the dashboard left snapshot', () => {
  const tree = renderer.create(<Provider store={store}><DashboardLeft /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
