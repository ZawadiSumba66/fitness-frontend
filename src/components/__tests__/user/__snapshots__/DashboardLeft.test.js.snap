// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the dashboard left snapshot 1`] = `
<div
  className="d-flex justify-content-center align-items-center spinner__wrapper"
>
  <div
    className="spinner-border text-primary"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
